### PR TITLE
Better inventory management for machines

### PR DESCRIPTION
**Pull request by vhd**
- fixes double chest access
 - adds support for "Linked Storage" mod

Basically SlotConfig now behave more like a hopper.

### CODE REVIEW

File: src/main/java/reborncore/api/items/InventoryUtils.java [hunk 1]
```diff
@@ -58,29 +67,46 @@ public static ItemStack insertItemStacked(Inventory inventory, ItemStack input,
 		return stack;
 	}
 
-	public static ItemStack insertItem(ItemStack input, BlockEntity blockEntity, Direction direction){
+	public static ItemStack insertItem(ItemStack input, Inventory inventory, Direction direction) {
```
Comment (modmuss50) [inline src/main/java/reborncore/api/items/InventoryUtils.java:70]: What’s the reasoning behind changing to an inv here? Might not be a bad idea to add the BE version back? I have a feeling the BE version is used.

Comment (vhd) [reply src/main/java/reborncore/api/items/InventoryUtils.java:70]: My reasons are:
 - the method itself expects Inventory
 - after rewrite i need a method that accepts Inventory directly 
 - all reborn machines implemented Inventory
 - after 2 hours of play testing it works just fine
**But** i’ll double check all usages soon and then let you know here

Comment (vhd) [reply src/main/java/reborncore/api/items/InventoryUtils.java:70]: Just checked. This function only used in SlotConfig (here), which is nice actually.

**Comment by vhd** [comment]
This will also partially fix https://github.com/TechReborn/TechReborn/issues/1941
We wan't be able to fix it completely.
Since we can transfer more than 1 item at a time stockpile will raise an error "lost $remainder items in a transaction"
https://github.com/branchpanic/stockpile/blob/c0752ce588c7e97bb0ac1fb50d65976eb6509e71/src/main/kotlin/me/branchpanic/mods/stockpile/impl/attribute/FixedMassItemInv.kt#L73

**Comment by modmuss50** [comment]
Seems better, thanks.

**Pull request merged**